### PR TITLE
STEP-100

### PR DESCRIPTION
**Pull request by dephillipsmichael**
Autofill country code and allow “+” characters to pass the regex.

### CODE REVIEW

**Comment by dephillipsmichael** [comment]
@EricSiegNW @Erin-Mounts for your review

File: mPower2/mPower2/Base.lproj/SignIn.json [hunk 1]
```diff
@@ -39,7 +39,7 @@
                                 "type":"string",
                                 "textFieldOptions":{
                                     "keyboardType":"phonePad",
-                                    "textValidator":"^\\d{10}$",
+                                    "textValidator":"^.{10,}$",
```
Comment (Erin-Mounts) [inline mPower2/mPower2/Base.lproj/SignIn.json:42]: so, [this site](https://www.howtocallabroad.com/netherlands/) says NL phone numbers are nine digits (not counting the country code prefix) and also [the Wikipedia article](https://en.wikipedia.org/wiki/Telephone_numbers_in_the_Netherlands) says "Non-geographical numbers have no fixed length" (non-geographical includes mobile numbers plus some other types) so I think we may just need to abandon the idea of validating phone numbers with a regex.

File: mPower2/mPower2/SignInTaskViewController.swift [hunk 2]
```diff
@@ -53,10 +53,22 @@ class SignInTaskViewController: RSDTaskViewController, SignInDelegate {
...
+    public static let NETHERLANDS_COUNTRY_CODE = "+31"
     
     /// The ISO country code for the region the user's phone number is in, defaults to US
     var regionCode: String? = US_REGION_CODE
+    
+    var countryCode: String? {
+        switch regionCode {
+        case SignInTaskViewController.NETHERLANDS_REGION_CODE:
+            return SignInTaskViewController.NETHERLANDS_COUNTRY_CODE
+        default:
```
Comment (Erin-Mounts) [inline mPower2/mPower2/SignInTaskViewController.swift:68]: Do we want to plan ahead and include provisions for Nigeria phone numbers now?

**Review by Erin-Mounts — APPROVED** [review]
See my comment re: continuing to use a regex to validate the number. Other than that I'm fine with this so will "pre-approve"

File: mPower2/mPower2/Base.lproj/SignIn.json (hunk 1 above)
Comment (dephillipsmichael) [reply mPower2/mPower2/Base.lproj/SignIn.json:42]: That regex I found is 10 any characters so it included the "+31" making NL 12 chars, but I agree there is really no reason to confuse things with a regex here.  I'll remove.

Comment (dephillipsmichael) [reply mPower2/mPower2/Base.lproj/SignIn.json]: Fixed and pushed

**Comment by Erin-Mounts** [comment]
Just thinking, while you're at it should we maybe change the alert that pops up when you get a 400 back from Bridge to not specify you need to enter a valid *US* phone number?

**Comment by dephillipsmichael** [comment]
> Just thinking, while you're at it should we maybe change the alert that pops up when you get a 400 back from Bridge to not specify you need to enter a valid _US_ phone number?
> 
> (PhoneRegistrationViewController.swift line 114)

That's a good idea, I'll change it to something more generic. How is this?

Title = "Wrong Number"
Msg = "Please enter a valid phone number."

**Comment by Erin-Mounts** [comment]
> > Just thinking, while you're at it should we maybe change the alert that pops up when you get a 400 back from Bridge to not specify you need to enter a valid _US_ phone number?
> > (PhoneRegistrationViewController.swift line 114)
> 
> That's a good idea, I'll change it to something more generic. How is this?
> 
> Title = "Wrong Number"
> Msg = "Please enter a valid phone number."

👍

**Pull request merged**